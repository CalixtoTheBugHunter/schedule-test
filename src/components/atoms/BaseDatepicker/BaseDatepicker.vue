<template>
    <div class="BaseDatepicker">
        <label v-if="!!label" class="BaseDatepicker__label">
            {{ label }}
        </label>
        <datepicker
            v-model="datepickerModel"
            v-bind="$attrs"
        />
    </div>
</template>

<script>
// Documentation here - https://icehaunter.github.io/vue3-datepicker/
import Datepicker from 'vue3-datepicker'

export default {
    name: 'BaseDatepicker',
    components: { Datepicker },
    props: {
        label: {
            type: String,
            default: ''
        },
        value: {
            type: [String, Number],
            default: null
        }
    },
    data() {
        return {
            datepickerModel: this.value
        }
    }
}
</script>

<style lang="scss" scoped>
.BaseDatepicker {
    color: $color-primary;
    &__label {
        color: $color-light;
        font-size: 16px;
        transition: transform 0.2s;
        pointer-events: none;
    }
}
</style>
