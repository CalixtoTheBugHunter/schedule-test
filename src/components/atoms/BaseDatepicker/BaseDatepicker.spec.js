/* eslint-disable no-undef */
import { shallowMount } from '@vue/test-utils'
import BaseDatepicker from './BaseDatepicker'

describe('BaseDatepicker', () => {
  const WRAPPER = shallowMount(BaseDatepicker)

  it('Component exists', () => {
    expect(WRAPPER.exists()).toBe(true)
  })

  it('Component not changed', () => {
    expect(WRAPPER.html()).toMatchSnapshot()
  })
})