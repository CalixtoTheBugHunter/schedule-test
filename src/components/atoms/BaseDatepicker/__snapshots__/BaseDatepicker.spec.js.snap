// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseDatepicker Component not changed 1`] = `
<div class="BaseDatepicker">
  <!--v-if-->
  <datepicker-stub></datepicker-stub>
</div>
`;
